fluidsynth/src/sfloader/fluid_defsfont.c

```diff
@@ -3190,26 +3192,32 @@ fixup_sample (SFData * sf)
 
 	  return (OK);
 	}
-      else if (inv_start || inv_end)
+      else if (invalid_loopstart || invalid_loopend || loopend_end_mismatch)
 	{
           /* loop is fowled?? (cluck cluck :) */
           invalid_loops |= TRUE;
 	  
 	  /* force incorrect loop points into the sample range, ignore padding */
-          if(inv_start)
+          if(invalid_loopstart)
 	  {
-            FLUID_LOG (FLUID_DBG, _("Sample '%s' has invalid loop start '%d',"
+            FLUID_LOG (FLUID_DBG, _("Sample '%s' has unusable loop start '%d',"
               " setting to sample start at '%d'+1"), sam->name, sam->loopstart, sam->start);
             sam->loopstart = sam->start + 1;
 	  }
           
-          if(inv_end)
+          if(invalid_loopend)
 	  {
-            FLUID_LOG (FLUID_DBG, _("Sample '%s' has invalid loop stop '%d',"
+            FLUID_LOG (FLUID_DBG, _("Sample '%s' has out-of-range loop stop '%d',"
               " setting to sample stop at '%d'-1"), sam->name, sam->loopend, sam->end);
             /* since sam->end points after valid sample data, set loopend to last sample available */
             sam->loopend = sam->end - 1;
```

Comment:
loopend has the same semantic as end, I think. so setting it to end-1 means that the loop reaches end-2 and then jumps to loopstart. I think it should be sam->loopend = sam->end here.